components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -673,20 +674,19 @@ void FeedV2Builder::BuildChannelFeed(const SubscriptionsSnapshot& subscriptions,
                 continue;
               }
 
-              auto publisher_it = info.publishers().find(
-                  item->get_article()->data->publisher_id);
-              if (publisher_it == info.publishers().end()) {
+              auto* publisher = base::FindOrNull(
+                  info.publishers(), item->get_article()->data->publisher_id);
+              if (!publisher) {
                 continue;
               }
 
               const auto& locale = info.locale();
 
-              auto locale_info_it =
-                  std::ranges::find_if(publisher_it->second->locales,
-                                       [&locale](const auto& locale_info) {
-                                         return locale == locale_info->locale;
-                                       });
-              if (locale_info_it == publisher_it->second->locales.end()) {
+              auto locale_info_it = std::ranges::find_if(
+                  (*publisher)->locales, [&locale](const auto& locale_info) {
```

Comment:
This doesn't look right. I suspect you were supposed to `FindPtrOrNull` here, rather than `FindOrNull`.

Reply:
66bc3c4376a300f556861a3bdb92504786904d4f